design_notes/XDN03 - xlang Type System.md

```diff
@@ -15,3 +15,142 @@ status: draft
 This design note describes the xlang's [type system](https://en.wikipedia.org/wiki/Type_system).
 The type system is designed to interoperate with a variety of object-oriented programming
 languages in a natural and familiar way.
+
+## Namespaces
+
+A namespace is a naming scope used to organize code and avoid naming collisions. All type
+categories in the xlang type system (enums, structs, delegates, interfaces, and runtime classes)
+except for fundamental types live in a namespace.
+
+Namespaces may contain other namespaces.
+
+xlang namespaces and type names are case preserving but insensitive. This means that you cannot
+have namespaces or type names that vary only by case. For example: you cannot have `Foo.SomeType`
+and `foo.SomeType` nor can you have `Foo.SomeType` and `Foo.someType`. However, you can have
+`Foo.SomeType` and `foo.AnotherType`. In this case `SomeType` and `AnotherType` would be considered
+as belonging to the same namespace.
+
+## Fundamental Types
+
+The xlang Type system includes a core set of built-in primitive types.
+
+xlang Type | Type Description
+---------- | ----------------------------------------------------------
+Int16      | a 16 bit signed integer
+Int32      | a 32 bit signed integer
+Int64      | a 64 bit signed integer
+UInt8      | an 8 bit unsigned integer
```

Comment:
There have been requests for Int8. https://github.com/Microsoft/xlang/issues/65

Reply:
We should add Int8. It's very useful for Machine Learning APIs

Reply:
will do